### PR TITLE
Update SetRandomHostPlugin.php

### PR DESCRIPTION
**Pull request by dbu**
a couple of cleanups and some questions / requests.

### CODE REVIEW

File: src/SetRandomHostPlugin.php [hunk 1]
```diff
@@ -19,30 +19,40 @@
...
 
     public function handleRequest(RequestInterface $request, callable $next, callable $first): Promise
     {
+        // would it make sense to cache this? either build all uri in the constructor, or just the one we currently use (in the happy case we will keep using the same)
+        // doing in the constructor would have the added benefit that we validate that the host names can be parsed.
         $uri = $this->uriFactory->createUri($this->hosts[$this->currentHostIndex]);
 
         $request = $request->withUri($request->getUri()
             ->withHost($uri->getHost())
+            // what happens if the uri does not specify a scheme or port?
```
Comment (ostrolucky) [inline src/SetRandomHostPlugin.php:55]: I don't think there is much we can do about that. We don't know if value we get from uriFactory is default of that factory or is coming from configured hosts. I'll use some URLs in test case though to make sure it works without, though

Comment (ostrolucky) [reply src/SetRandomHostPlugin.php]: Ah uriFactory returns empty host in those cases. Addressed.

File: src/SetRandomHostPlugin.php [hunk 2]
```diff
@@ -5,41 +5,59 @@
...
 {
-    private UriFactoryInterface $uriFactory;
-    /** @var list<string> */
-    private array $hosts;
+    /**
+     * A list of base URLs
+     *
+     * Only protocol, host and port are used. Paths should not be set and are ignored.
+     *
+     * @var list<string>
```
Comment (dbu) [inline src/SetRandomHostPlugin.php:28]:
```suggestion
     * @var UriInterface[]
```

File: src/SetRandomHostPlugin.php [hunk 3]
```diff
@@ -5,41 +5,59 @@
 namespace Http\Client\Common\Plugin;
 
 use Http\Client\Common\Plugin;
-use Http\Client\Exception\NetworkException;
 use Http\Promise\Promise;
 use Psr\Http\Client\ClientExceptionInterface;
+use Psr\Http\Client\NetworkExceptionInterface;
 use Psr\Http\Message\RequestInterface;
 use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\UriFactoryInterface;
```
Comment (dbu) [inline src/SetRandomHostPlugin.php:13]:
```suggestion
use Psr\Http\Message\UriFactoryInterface;
use Psr\Http\Message\UriInterface;
```

**Comment by ostrolucky** [comment]
changes merged separately, thx for review!

**Pull request closed without merging**